src/uploaders.rs

```diff
@@ -90,17 +100,28 @@ impl Uploader {
     ///
     /// This function can panic on an `Self::Local` during development.
     /// Production and tests use `Self::S3` which should not panic.
-    pub fn upload<R: std::io::Read + Send + 'static>(
+    pub fn upload<R: std::io::Read + std::io::Seek + Send + 'static>(
         &self,
         client: &Client,
         path: &str,
         mut content: R,
-        content_length: u64,
         content_type: &str,
         extra_headers: header::HeaderMap,
+        upload_bucket: UploadBucket,
     ) -> Result<Option<String>> {
         match *self {
-            Uploader::S3 { ref bucket, .. } => {
+            Uploader::S3 {
+                ref bucket,
+                ref index_bucket,
+                ..
+            } => {
+                let bucket = match upload_bucket {
+                    UploadBucket::Index => index_bucket,
+                    UploadBucket::Default => bucket,
+                };
+                let content_length = content.seek(SeekFrom::End(0))?;
+                content.seek(SeekFrom::Start(0))?;
```

Comment:
what's the reason for the `content_length` change here? on first glance it looks like a slightly unrelated change, which we'd ideally merge and deploy separately to lower the risks. does that make sense?

Reply:
I can move it into a separate change, but it will have to go in before this one to keep clippy happy.

Reply:
> clippy said there were too many arguments

I see, thanks :)